core/src/main/java/com/linecorp/armeria/internal/server/annotation/AnnotatedDocServicePlugin.java

```diff
@@ -207,10 +207,19 @@ private static String normalizeParameterized(Route route) {
 
         final StringBuilder sb = new StringBuilder();
         for (String paramName : route.paramNames()) {
-            final int colonIndex = path.indexOf(':', beginIndex);
+            int colonIndex = path.indexOf(':', beginIndex);
+            boolean restPathsPattern = false;
+            if (colonIndex == -1) {
+                colonIndex = path.indexOf('*', beginIndex);
+                restPathsPattern = colonIndex != -1;
+            }
             assert colonIndex != -1;
```

Comment:
Is `restPathsPattern` always `true` here inside `if (colonIndex == -1) { ... }`? If so, this could be simplified into:
```suggestion
            if (colonIndex == -1) {
                colonIndex = path.indexOf('*', beginIndex);
                assert colonIndex != -1;
                restPathsPattern = true;
            }
```